x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -585,9 +585,12 @@ export function updateLayerIndexPattern(
  */
 export function getErrorMessages(layer: IndexPatternLayer): string[] | undefined {
   const errors: string[] = [];
-
   Object.entries(layer.columns).forEach(([columnId, column]) => {
-    const def = operationDefinitionMap[column.operationType];
+    // If we're transitioning to another operation, check for "new" incompleteColumns rather
+    // than "old" saved operation on the layer
+    const columnFinalRef =
+      layer.incompleteColumns?.[columnId]?.operationType || column.operationType;
+    const def = operationDefinitionMap[columnFinalRef];
```

Comment:
I think this logic could cause a problem if the `getErrorMessage` code looks at the current column. For example, if the function _thinks_ it's a reference-based column and tries to read `.references.every` but it's the wrong type.

What if we skipped cases where there is an `incompleteColumn`, because we can't get errors the right way?